### DISCUSSION TITLE
Improve the Text edit controls for Shell commands

### DISCUSSION THREAD

**Discussion by XXXBold** [borgbase/vorta#847] [Ideas]
**Is your feature request related to a problem? Please describe.**
Not really a problem, but an inconvenience. If I want to use longer/more complex pre/post backup shell commands, the content gets lost very quick due to the single-line text control.

For Excample, the commandline I have now looks like this:
`tar -cjf "$(dirname "$repo_url")/($profile_name)_vortabackup.tar" -C "$(dirname "$repo_url")" "$(basename "$repo_url")"`

I will probably make it longer, but it's barely readable already (With the default window size, without expanding the window, but that will not solve, only delay the problem when it gets longer).

**Describe the solution you'd like**
Change the single-line text control to multiline.

***Implementation Example***
Show a single line as default option (as it's now) when no shell command is present.
If the user edits the commandline and hits the return key, expand (grow) the control one line down. On the other hand, if the user deletes a complete line, shrink the control accordingly, so the control basically grows and shrinks with the text.

**Describe alternatives you've considered**
- Use an external text edior to create the commandline and paste it
- Create an external shellscript and call it from the builtin commandline. 

The proposed feature is superior to the alternatives imho.

**Comment by m3nu** [comment]
Yeah. Could make sense.

**Comment by samuel-w** [comment]
Found a relevant stackoverflow answer: https://stackoverflow.com/a/13599165. We could call `text_edit.setFixedHeight(text_edit.document().size().height() + text_edit.contentsMargins().top()*2)` on every key input
